File: platform/polyglot/rag_llm_synthesis/mjs/src/providers/anthropic.mjs
/**
 * @fileoverview Anthropic provider — lazy-loaded.
 */

let _client = null;

/**
 * Get or create a lazy-initialized Anthropic client.
 * Requires the `@anthropic-ai/sdk` npm package.
 * @returns {Promise<any>}
 */
export async function getAnthropicClient() {
  if (!_client) {
    try {
      const { default: Anthropic } = await import('@anthropic-ai/sdk');
      _client = new Anthropic();
    } catch {
      throw new Error('LLM provider is not available. Required dependency is not installed.');
    }
  }
  return _client;
}

export function resetClient() {
  _client = null;
}

File: platform/polyglot/rag_llm_synthesis/py/rag_llm_synthesis/providers/anthropic.py
"""Anthropic provider client factory."""

from __future__ import annotations


_client = None


def get_anthropic_client():
    """Get or create a lazy-initialized Anthropic client."""
    global _client
    if _client is None:
        try:
            import anthropic
        except ImportError:
            raise RuntimeError("LLM provider is not available. Required dependency is not installed.")
        _client = anthropic.Anthropic()
    return _client


def reset_client():
    """Reset the cached client (for testing)."""
    global _client
    _client = None


_async_client = None


def get_async_anthropic_client():
    """Get or create a lazy-initialized async Anthropic client."""
    global _async_client
    if _async_client is None:
        try:
            import anthropic
        except ImportError:
            raise RuntimeError("LLM provider is not available. Required dependency is not installed.")
        _async_client = anthropic.AsyncAnthropic()
    return _async_client


def reset_async_client():
    """Reset the cached async client (for testing)."""
    global _async_client
    _async_client = None
